sound/soc/sof/sof-audio.c

```diff
@@ -615,42 +615,6 @@ bool snd_sof_stream_suspend_ignored(struct snd_sof_dev *sdev)
 	return false;
 }
 
-int sof_set_hw_params_upon_resume(struct device *dev)
-{
-	struct snd_sof_dev *sdev = dev_get_drvdata(dev);
-	struct snd_pcm_substream *substream;
-	struct snd_sof_pcm *spcm;
-	snd_pcm_state_t state;
-	int dir;
-
-	/*
-	 * SOF requires hw_params to be set-up internally upon resume.
-	 * So, set the flag to indicate this for those streams that
-	 * have been suspended.
-	 */
-	list_for_each_entry(spcm, &sdev->pcm_list, list) {
-		for_each_pcm_streams(dir) {
-			/*
-			 * do not reset hw_params upon resume for streams that
-			 * were kept running during suspend
-			 */
-			if (spcm->stream[dir].suspend_ignored)
-				continue;
-
-			substream = spcm->stream[dir].substream;
-			if (!substream || !substream->runtime)
-				continue;
-
-			state = substream->runtime->status->state;
-			if (state == SNDRV_PCM_STATE_SUSPENDED)
-				spcm->prepared[dir] = false;
-		}
-	}
-
-	/* set internal flag for BE */
-	return snd_sof_dsp_hw_params_upon_resume(sdev);
-}
-
```

Comment:
The change is probably ok, but the commit message does not even being to describe why this change is needed.

I think you would need to explain that we introduced a set of new helpers in 'ASoC: SOF: Add a helper for freeing PCM stream' which invalidated the previous logic wrt. the prepared flag, and I think a Fixes: flag is needed since this is really re-enabling functionality that was working previously and broken by changes.

If I am mistaken on why this change is needed, it's even more of a reason to explain it :-)

Reply:
and your commit message would need to have a 

````
BugLink: https://github.com/thesofproject/linux/issues/3543
````

Reply:
ok i've updated the commit message now. let me know if it reads better